app/policies/api/v1/offering_policy.rb

```diff
@@ -0,0 +1,26 @@
+class API::V1::OfferingPolicy < ApplicationPolicy
+
+  def show?
+    is_class_teacher_or_admin?
+  end
+
+  def collaborators_data?
+    request_is_peer?
+  end
+
+  private
+  def is_admin?
+    user && user.has_role?("admin")
+  end
+
+  def is_class_teacher?
+    return false unless user && user.portal_teacher
+    offering = Portal::Offering.find_by_id(params[:id])
+    offering && offering.clazz.teachers.include?(user.portal_teacher)
```

Comment:
Instead of looking up the offering here you should let the controller do that.
So in the controller you do `authorize @offering`, then in the policy methods you can access that passed in offering with the variable `record`.